Upgrade ui-lib to latest Stencil version

scottopherson opened the pull request:
### :nut_and_bolt: Description

This PR upgrades `chef-ui-library`'s Stencil and Jest dependencies to their latest version. Includes fixes for any breaking changes, deprecation warnings, etc.

**Before:**

```
$ npm outdated

Package           Current  Wanted   Latest  Location
@stencil/core      0.18.1  0.18.1    1.1.7  chef-ui-library
@stencil/router     0.2.1   0.2.8    1.0.1  chef-ui-library
@stencil/sass       0.1.1   0.1.1    1.0.1  chef-ui-library
@types/jest        23.3.9  23.3.9  24.0.15  chef-ui-library
@types/puppeteer   1.11.2  1.11.2   1.12.4  chef-ui-library
jest               23.6.0  23.6.0   24.8.0  chef-ui-library
jest-cli           23.6.0  23.6.0   24.8.0  chef-ui-library
puppeteer          1.11.0  1.11.0   1.18.1  chef-ui-library
```
```
$ npm audit

found 284 vulnerabilities (248 low, 1 moderate, 35 high) in 40052 scanned packages
```

**After:**

```
$ npm outdated

Package    Current  Wanted  Latest  Location
puppeteer   1.17.0  1.17.0  1.19.0  chef-ui-library
```
**Stencil has [a hard dependency](https://github.com/ionic-team/stencil/issues/1360) on `puppeteer 1.17.0`*

```
$ npm audit

found 0 vulnerabilities in 1748741 scanned packages
```

### :+1: Definition of Done

`chef-ui-library` is using latest version of Stencil and Jest.

### :chains: Related Resources

https://github.com/chef/automate/issues/186
https://github.com/ionic-team/stencil/blob/master/CHANGELOG.md
https://github.com/ionic-team/stencil/blob/master/BREAKING_CHANGES.md

### :white_check_mark: Checklist

- [x] Necessary tests added/updated?
- [x] Necessary docs added/updated?
- [x] Code actually executed?
- [x] Vetting performed (unit tests, lint, etc.)?

scottopherson commented:
So these commits give us a successful upgrade to latest version of Stencil. All tests pass and so far I haven't noticed any regressions in my manual testing of the UI. That said, there are some new warnings currently being outputted from the Stencil build:

```
$ stencil build

[08:59.2]  @stencil/core v1.1.9 🐕
[08:59.3]  build, chef, prod mode, started ...
[08:59.4]  transpile started ...
[09:04.3]  transpile finished in 4.94 s
[09:04.3]  type checking started ...
[09:04.3]  copy started ...
[09:04.3]  generate styles started ...
[09:04.4]  bundling components started ...
[09:05.7]  copy finished (40 files) in 1.35 s
[09:08.3]  generate styles finished in 3.98 s
[09:17.3]  type checking finished in 12.96 s
[09:25.6]  bundling components finished in 21.22 s

[ WARN  ]  Build Warn: src/atoms/chef-badge/chef-badge.tsx:56:34
           The @Prop() name "id" is a reserved public name. Please rename the "id" prop so it does not conflict with an
           existing standardized prototype member. Reusing prop names that are already defined on the element's
           prototype may cause unexpected runtime errors or user-interface issues on various browsers, so it's best to
           avoid them entirely.

     L55:     */
     L56:    @Prop({ reflectToAttr: true }) id: string;

[ WARN  ]  Build Warn: src/atoms/chef-checkbox/chef-checkbox.tsx:37:12
           The event name conflicts with the "change" native DOM event name.

     L37:    @Event() change: EventEmitter;

[ WARN  ]  Build Warn: src/atoms/chef-snippet/chef-snippet.tsx:53:11
           The @Prop() name "lang" is a reserved public name. Please rename the "lang" prop so it does not conflict
           with an existing standardized prototype member. Reusing prop names that are already defined on the element's
           prototype may cause unexpected runtime errors or user-interface issues on various browsers, so it's best to
           avoid them entirely.

     L52:     */
     L53:    @Prop() lang = 'html';

[ WARN  ]  Build Warn: src/charts/chef-radial-chart/chef-radial-chart.tsx:87:11
           The @Prop() name "id" is a reserved public name. Please rename the "id" prop so it does not conflict with an
           existing standardized prototype member. Reusing prop names that are already defined on the element's
           prototype may cause unexpected runtime errors or user-interface issues on various browsers, so it's best to
           avoid them entirely.

     L86:     */
     L87:    @Prop() id: string;

[ WARN  ]  Build Warn: src/molecules/chef-combobox/chef-combobox.tsx:68:12
           The event name conflicts with the "change" native DOM event name.

     L68:    @Event() change: EventEmitter;

[ WARN  ]  Build Warn: src/molecules/chef-control-menu/chef-control-menu.tsx:88:12
           The event name conflicts with the "change" native DOM event name.

     L87:     */
     L88:    @Event() change: EventEmitter;

[ WARN  ]  Build Warn: src/molecules/chef-radio/chef-radio.tsx:28:12
           The event name conflicts with the "change" native DOM event name.

     L28:    @Event() change: EventEmitter;
     L29:    @Event() input: EventEmitter;

[ WARN  ]  Build Warn: src/molecules/chef-radio/chef-radio.tsx:29:12
           The event name conflicts with the "input" native DOM event name.

     L28:    @Event() change: EventEmitter;
     L29:    @Event() input: EventEmitter;

[ WARN  ]  Build Warn: src/molecules/chef-phat-radio/chef-phat-radio.tsx:28:12
           The event name conflicts with the "change" native DOM event name.

     L28:    @Event() change: EventEmitter;
     L29:    @Event() input: EventEmitter;

[ WARN  ]  Build Warn: src/molecules/chef-phat-radio/chef-phat-radio.tsx:29:12
           The event name conflicts with the "input" native DOM event name.

     L28:    @Event() change: EventEmitter;
     L29:    @Event() input: EventEmitter;

[ WARN  ]  Build Warn: src/molecules/chef-select/chef-select.tsx:101:12
           The event name conflicts with the "change" native DOM event name.

    L100:     */
    L101:    @Event() change: EventEmitter;

[ WARN  ]  Build Warn: src/molecules/chef-status-filter-group/chef-status-filter-group.tsx:62:12
           The event name conflicts with the "change" native DOM event name.

     L62:    @Event() change: EventEmitter;
     L63:    @Event() input: EventEmitter;

[ WARN  ]  Build Warn: src/molecules/chef-status-filter-group/chef-status-filter-group.tsx:63:12
           The event name conflicts with the "input" native DOM event name.

     L62:    @Event() change: EventEmitter;
     L63:    @Event() input: EventEmitter;

[ WARN  ]  Build Warn: src/molecules/chef-tab-selector/chef-tab-selector.tsx:53:12
           The event name conflicts with the "change" native DOM event name.

     L53:    @Event() change: EventEmitter;
     L54:    @Event() input: EventEmitter;

[ WARN  ]  Build Warn: src/molecules/chef-tab-selector/chef-tab-selector.tsx:54:12
           The event name conflicts with the "input" native DOM event name.

     L53:    @Event() change: EventEmitter;
     L54:    @Event() input: EventEmitter;

[ WARN  ]  Build Warn: src/molecules/chef-toggle/chef-toggle.tsx:53:12
           The event name conflicts with the "change" native DOM event name.

     L53:    @Event() change: EventEmitter;
     L54:    @Event() input: EventEmitter;

[ WARN  ]  Build Warn: src/molecules/chef-toggle/chef-toggle.tsx:54:12
           The event name conflicts with the "input" native DOM event name.

     L53:    @Event() change: EventEmitter;
     L54:    @Event() input: EventEmitter;

[09:25.9]  build finished in 26.65 s
```

We need to rename some of the event and property names of our ui-lib components to suppress these warnings. Which means we'll need to discuss how we want to rename these properties and then make the necessary code changes. Doing so will then require updating all usages of these ui-lib components within `automate-ui` to use the new naming scheme we've chosen. Since this PR currently upgrades `chef-ui-library` without requiring any changes within `automate-ui`, I think I'd like to address these naming warnings in a separate PR to keep this one from getting even larger. Thoughts? Either way, this branch should be good for y'all to pull down and test out.

msorens commented on components/chef-ui-library/src/app.tsx:
```diff
@@ -2,10 +2,10 @@
 
-import { Component, Prop } from '@stencil/core';
-import { Store } from '@stencil/redux';
+import { Component, Prop, h } from '@stencil/core';
```
What is this mysterious `h` I see everywhere?

scottopherson replied:
https://github.com/ionic-team/stencil/blob/master/BREAKING_CHANGES.md#import--h--is-required

scottopherson commented:
@msorens not quite sure; my eyebrow raised as well. We didn't add or remove any dependencies of our own so I can only assume something changed with the dependencies that our direct dependencies depend on to cause that number to jump up so high.

scottopherson commented:
`typedoc 0.15.0` was released today so I've added a commit to bump to this version. This addresses the remaining `npm audit` warnings:

```
$ npm audit

found 0 vulnerabilities in 1748741 scanned packages
```

Added a commit to bump to today's Stencil release as well. `1.2.0 -> 1.2.2`
